src/main/java/edu/harvard/iq/dataverse/AbstractIdServiceBean.java

```diff
@@ -28,68 +28,70 @@ public String getIdentifierForLookup(String protocol, String authority, String s
+        metadata.put("_target", getTargetUrl(dvObjectIn));
         return metadata;
     }
 
-    protected HashMap<String, String> getUpdateMetadataFromDataset(Dataset datasetIn) {
+    protected HashMap<String, String> getUpdateMetadata(DvObject dvObjectIn) {
         logger.log(Level.FINE,"getUpdateMetadataFromDataset");
         HashMap<String, String> metadata = new HashMap<>();
-        metadata = addBasicMetadata(datasetIn, metadata);
+        metadata = addBasicMetadata(dvObjectIn, metadata);
         return metadata;
     }
     
-    protected HashMap<String, String> addBasicMetadata(Dataset datasetIn, HashMap<String, String> metadata){
-        String authorString = datasetIn.getLatestVersion().getAuthorsStr();
+    protected HashMap<String, String> addBasicMetadata(DvObject dvObjectIn, HashMap<String, String> metadata) {
+
+        String authorString = dvObjectIn.getAuthorString();
 
         if (authorString.isEmpty()) {
             authorString = ":unav";
         }
 
         String producerString = dataverseService.findRootDataverse().getName();
 
-        if(producerString.isEmpty()) {
+        if (producerString.isEmpty()) {
             producerString = ":unav";
         }
+        
         metadata.put("datacite.creator", authorString);
-        metadata.put("datacite.title", datasetIn.getLatestVersion().getTitle());
+        metadata.put("datacite.title", dvObjectIn.getDisplayName());
         metadata.put("datacite.publisher", producerString);
-        
-        
         return metadata;
-    }
-
-    @Override
-    public HashMap<String, String> getMetadataFromDatasetForTargetURL(Dataset datasetIn) {
-        logger.log(Level.FINE,"getMetadataFromDatasetForTargetURL");
-        HashMap<String, String> metadata = new HashMap<>();
-        metadata.put("_target", getTargetUrl(datasetIn));
-        return metadata;
-    }
+    }   
 
-    protected String getTargetUrl(Dataset datasetIn) {
+    protected String getTargetUrl(DvObject dvObjectIn) {
         logger.log(Level.FINE,"getTargetUrl");
-        return systemConfig.getDataverseSiteUrl() + Dataset.TARGET_URL + datasetIn.getGlobalId();
+        return systemConfig.getDataverseSiteUrl() + dvObjectIn.getTargetUrl() + dvObjectIn.getGlobalId();
     }
-
+    
     @Override
-    public String getIdentifierFromDataset(Dataset dataset) {
-        logger.log(Level.FINE,"getIdentifierFromDataset");
-        return dataset.getGlobalId();
+    public String getIdentifier(DvObject dvObject) {
+        return dvObject.getGlobalId();
     }
     
-    protected String generateYear (Dataset datasetIn){
-        if (datasetIn.isReleased()) {
-            return datasetIn.getPublicationDateFormattedYYYYMMDD().substring(0, 4);
-        }
-        return new SimpleDateFormat("yyyy").format(datasetIn.getCreateDate()); 
+    protected String generateYear (DvObject dvObjectIn){
+        return dvObjectIn.getYearPublishedCreated(); 
+    }
+    
+     @Override
```

Comment:
This implementation should probably be removed. The bean is abstract anyway, and having a method you can't call is not very helpful.